### PR TITLE
Add `export_parameters` to metadata.json in export archives

### LINKED ISSUE #3360: Add Link rules to metadata.json in export archive
In order to:
- better help debugging export/import issues and
- keep information concerning _how_ the archive was created,

I suggest we add a small dictionary to the archive's `metadata.json` file that simply states the rules used to generate the particular archive.
This may look something like:

```jsonc
{
  "link_rules": {
    "input_calc_forward": false,
    "input_calc_backward": true,
    "create_forward": true,
    "create_backward": true,
    "return_forward": true,
    "return_backward": false,
    "input_work_forward": false,
    "input_work_backward": true,
    "call_calc_forward": true,
    "call_calc_backward": false,
    "call_work_forward": true,
    "call_work_backward": false
  },
  // ...
}
```

### PR DESCRIPTION
**Pull request by CasperWA**
Closes #3360 

Current "data structure":
```jsonc
{
  "export_parameters": {
    "link_follow_rules": {
      // Updated dict from config.LINK_FLAGS
    },
    "original_entities": {
      "Node": [
        // UUIDs of originally chosen Nodes that should be exported
      ],
      // Possibly also "Computer": [] and "Group": [].
      // "Node": [] May also be left out, if no Nodes were specified to be exported.
    },
    "licence_info": {
      "allowed_licenses": null, // or a list or, if the parameter was a function, a string, naming the function that was used
      "forbidden_licenses": null // same as for "allowed_licenses"
    },
    "include_comments": true, // reflects the boolean parameter of the same name
    "include_logs": true // reflects the boolean parameter of the same name
  }
}
```

Example, which is also added to the docs:
```jsonc
{
  "export_parameters": {
    "link_follow_rules": {
      "input_calc_forward': False,
      "input_calc_backward': True,
      "create_forward': True,
      "create_backward': True,
      "return_forward': True,
      "return_backward': False,
      "input_work_forward': False,
      "input_work_backward': True,
      "call_calc_forward': True,
      "call_calc_backward': False,
      "call_work_forward': True,
      "call_work_backward': False
    },
    "original_entities": {
      "Node": ["1024e35e-166b-4104-95f6-c1706df4ce15"]
    },
    "license_info": {
      "allowed_licenses": None,
      "forbidden_licenses": "Function valid_licenses was used to check for forbidden licenses"
    },
    "include_comments": True,
    "include_logs": False
  },
  // ...
}  
```

### CODE REVIEW

File: aiida/tools/importexport/dbexport/__init__.py [hunk 1]
```diff
@@ -425,22 +435,35 @@ def export_tree(
...
+            # The value is a function (can either be a function, a list or None)
+            license_info[name] = 'Function {} was used to check for {}'.format(value, name.replace('_', ' '))
+
     metadata = {
         'aiida_version': get_version(),
         'export_version': EXPORT_VERSION,
         'all_fields_info': all_fields_info,
         'unique_identifiers': unique_identifiers,
+        'export_parameters': {
+            'link_follow_rules': link_follow_rules,
```
Comment (sphuber) [inline aiida/tools/importexport/dbexport/__init__.py]: I'd suggest to rename `link_follow_rules` to `graph_traversal_rules` to keep the naming with the `GraphTraversalRule` constants in `aiida.common.links`

File: aiida/tools/importexport/dbexport/__init__.py [hunk 2]
```diff
@@ -425,22 +435,35 @@ def export_tree(
...
+            'original_entities': original_entities,
```
Comment (sphuber) [inline aiida/tools/importexport/dbexport/__init__.py]: Maybe `entity_starting_set` would be a bit clearer instead of `original_entities`?

**Comment by CasperWA** [comment]
If/when tests succeed, this should be ready to go

File: docs/source/import_export/main.rst [hunk 3]
```diff
@@ -60,8 +60,33 @@ Let's have a look at the contents of ``metadata.json``:
 .. code-block:: json
 
     {
-      "export_version": "0.6",
+      "export_version": "0.7",
       "aiida_version": "1.0.0",
+      "export_parameters": {
```
Comment (sphuber) [inline docs/source/import_export/main.rst]: To be updated to recent changes

**Pull request merged**